vespalib/src/vespa/vespalib/net/tls/assumed_roles.h

```diff
@@ -0,0 +1,80 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <vespa/vespalib/stllike/hash_set.h>
+#include <vespa/vespalib/stllike/string.h>
+#include <vector>
+#include <iosfwd>
+
+namespace vespalib { class asciistream; }
+
+namespace vespalib::net::tls {
+
+/**
+ * Encapsulates a set of roles that requests over a particular authenticated
+ * connection can assume, based on the authorization rules it matched during mTLS
+ * handshaking.
+ *
+ * If at least one role is a wildcard ('*') role, the connection can assume _any_
+ * possible role. This is the default when no role constraints are specified in
+ * the TLS configuration file (legacy behavior). However, a default-constructed
+ * AssumedRoles instance does not allow any roles to be assumed.
+ */
+class AssumedRoles {
```

Comment:
consider Capabilities

Reply:
Will address renaming of concepts in a follow-up